### PR TITLE
[WIP] Style transfer

### PR DESCRIPTION
**Pull request by JunbinWang**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
TensorLayer implementation of the ICCV 2017 Paper [Arbitrary Style Transfer in Real-time with Adaptive Instance Normalization](https://arxiv.org/abs/1703.06868) which supports any styles in one single model.

### CODE REVIEW

File: example/adaptive_style_transfer/train.py [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+CONTENT_DATA_PATH = '/root/even/dataset/COCO_train_2014/'
+STYLE_DATA_PATH = '/root/even/dataset/wiki_all_images/'
+
+if __name__ == '__main__':
+
+    start_time = datetime.now()
+
+    # Get the path of all valid images
+    print('Preprocessing training images \n')
+    content_images = utils.list_images(CONTENT_DATA_PATH)
```
Comment (zsdonghao) [inline example/adaptive_style_transfer/train.py]: 这个可以用 `tl.files.load_file_list` 代替

File: tensorlayer/layers/image_resampling.py [hunk 2]
```diff
@@ -17,7 +17,6 @@
 
 class UpSampling2dLayer(Layer):
```
Comment (zsdonghao) [inline tensorlayer/layers/image_resampling.py]: 这个在  `tl.layers` 里面改呗。然后直接用 `tl.layers.UpSampling2dLayer` 就好了

File: example/adaptive_style_transfer/README.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
+# Tensorlayer Adaptive Style Transfer
+------
+
+1. Tensorlayer implement of the ICCV 2017 Paper [Arbitrary Style Transfer in Real-time with Adaptive Instance Normalization](https://arxiv.org/abs/1703.06868)
+
+2. You can use the  <b>train.py</b> script to train your own model.  Alternatively, you can use the <b>test.py</b> script to run my pretrained models.
+My pretrained models can be downloaded from [here](https://github.com/tensorlayer/pretrained-models/tree/master/models/style_transfer_pretrained_models), and put into the 'pretrained_models' folder for testing.
+
+3. To train the model, you need to download [MSCOCO dataset](http://cocodataset.org/#download) and [Wikiart dataset](https://www.kaggle.com/c/painter-by-numbers).
```
Comment (zsdonghao) [inline example/adaptive_style_transfer/README.md]: 下载之后放在哪里？

Comment (JunbinWang) [reply example/adaptive_style_transfer/README.md]: 我写了 'pretrained_models' folder for testing... 然后我本地建了这个空文件夹，传上去不知道为啥没了..
